Documentation/AdditionalFunctionality/PushNotifications.md

```diff
@@ -0,0 +1,133 @@
+# Push Notifications
+
+The `NotificationCenter` handles the registration and sending of **push notifications** to Apple Push Notification Service (APNS) and Firebase Cloud Messaging (FCM). APNS can send out push notifications to iOS, macOS and tvOS devices, while FCM handles messaging for iOS, Android and Web Push.
+
+## Configuration
+
+In order to send push notifications, the Apodini web server needs to be authenticated to at least one push notification service. These services are added to a web server by defining them in the `configuration` computed property.
+
+This example adds APNS to the web server.
+
+```swift
+var configuration: some Configuration {
+    APNSConfiguration(.pem(pemPath: "certificate.pem"), identifier: "de.tum.in.ase.Example", environment: .sandbox)
+}
+```
+
+## Usage
+
+The `NotificationCenter` can be injected to any `Component` by using the `@Environment` property wrapper.
+
+```swift
+struct NewsAlertComponent: Component {
+    @Environment(\.notificationCenter) var notificationCenter: NotificationCener
+
+    // ...
+}
+```
+
+## Device Registration
+
+Devices can be registered to the `NotificationCenter` by conforming to the `Device` protocol. This protocol consists of the following proerties:
+
+- **type**: This property specifies which push notification service to use.
+- **token**: The device id for one app used by the push notification service
+- **subscriptions**: Subscriptions are used to group `Device`s together. Example: All devices that want to receive news alerts for a specific topic.
+
+```swift
+public struct Device: Content {
+    public var type: PushNotificationService
+    public var token: String
+    public var subscriptions: [String]?
+}
+
+public enum PushNotificationService: String {
+    case apns
+    case fcm
+}
+```
+
+The web service needs a specific route and `Component` to handle push notification registration. The following example shows how to add a `Device` to the currently authenticated `User`.
+
+```swift
+struct RegisterComponent: Component {
+    @Request(\.user) var user: User  // The logged in user
```

Comment:
I think we agreed on keeping the number of property wrappers down if possible. Furthermore, `@Request` is protocol specific terminology. The way how the user is identified and authenticated could be based on requests and session-cookies or a WebSocket connection with respective authentication measures in place. I think using `@Environment(\.user)` would be better.

Reply:
Sorry, I just noticed that `@Request` is already documented. I still do not agree with the terminology, but this discussion should therefore not be part of this PR.